controllers/finalizers.go

```diff
@@ -57,12 +56,9 @@ func finalizeNodes(r *SpecialResourceReconciler, remove string) error {
 			return errors.Wrap(err, "forbidden check Role, ClusterRole and Bindings for operator %s")
 		}
 		if apierrors.IsConflict(err) {
-			var err error
-
 			if err = cache.Nodes(r.specialresource.Spec.NodeSelector, true); err != nil {
```

Comment:
I think we should do the following:
```suggestion
			if err := cache.Nodes(r.specialresource.Spec.NodeSelector, true); err != nil {
```
Otherwise if `cache.Nodes` returns `nil`, then `fmt.Errorf` below will wrap a `nil` error instead of the actual error.

Reply:
good catch, fixed